### PR TITLE
Migrate to typescript

### PR DESCRIPTION
**Pull request by thabarbados**
### Fix or Enhancement?
Add typescript and rewrite code

- [x] All tests passed

### Environment
- OS: BigSur MacOs
- NPM Version: 8.5.1

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -19,9 +21,9 @@ module.exports = {
...
     'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0,
     'prettier/prettier': 'error',
-    quotes: [2, 'single', 'avoid-escape'],
+    quotes: [2, 'single', 'avoid-escape']
   },
   globals: {
-    LIB_VERSION: true,
-  },
-};
+    LIB_VERSION: true
```
Comment (dangreen) [inline .eslintrc.js:27]: there is no global var with this name

**Review by dangreen — CHANGES_REQUESTED** [review]
- What about tests, stories, etc?
- Please remove https://github.com/apertureless/vue-chartjs/tree/main/types
- Let's add tests for types (e. g. https://github.com/reactchartjs/react-chartjs-2/blob/master/test/chart.test-d.tsx, https://github.com/SamVerschueren/tsd)

File: .eslintrc.js (hunk 1 above)
Comment (thabarbados) [reply .eslintrc.js]: Added Typescript to stories, sandboxes, tests. Added test for types definition. Removed unused vars and folders

File: .clean-publish [hunk 2]
```diff
@@ -1,4 +1,6 @@
 {
   "withoutPublish": true,
-  "tempDir": "package"
+  "tempDir": "package",
+  "fields": ["tsd"],
+  "files": ["website"]
```
Comment (dangreen) [inline .clean-publish:5]:
```suggestion
  "files": ["docs"]
```

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -26,3 +26,25 @@ jobs:
         uses: andresz1/size-limit-action@master
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
+  typings:
```
Comment (dangreen) [inline .github/workflows/checks.yml:29]: please add check for storybook https://github.com/reactchartjs/react-chartjs-2/blob/master/.github/workflows/checks.yml#L48

**Review by dangreen — CHANGES_REQUESTED** [review]

File: .clean-publish (hunk 2 above)
Comment (thabarbados) [reply .clean-publish]: fixed

File: .github/workflows/checks.yml (hunk 3 above)
Comment (thabarbados) [reply .github/workflows/checks.yml:29]: added

**Review by dangreen — APPROVED** [review]

**Pull request merged**